packages/backend-core/src/objectStore/objectStore.ts

```
@@ -105,37 +110,43 @@ export function ObjectStore(
  * Given an object store and a bucket name this will make sure the bucket exists,
  * if it does not exist then it will create it.
  */
-export async function makeSureBucketExists(client: any, bucketName: string) {
+export async function createBucketIfNotExists(
+  client: any,
+  bucketName: string
+): Promise<boolean> {
   bucketName = sanitizeBucket(bucketName)
   try {
     await client
       .headBucket({
         Bucket: bucketName,
       })
       .promise()
+    return false
   } catch (err: any) {
     const promises: any = STATE.bucketCreationPromises
     const doesntExist = err.statusCode === 404,
       noAccess = err.statusCode === 403
     if (promises[bucketName]) {
       await promises[bucketName]
+      return false
     } else if (doesntExist || noAccess) {
       if (doesntExist) {
-        // bucket doesn't exist create it
         promises[bucketName] = client
           .createBucket({
             Bucket: bucketName,
           })
           .promise()
         await promises[bucketName]
         delete promises[bucketName]
+        return true
+      } else {
+        throw new Error("Access denied to object store bucket.")
```

We should throw the original error, just in case we are hiding something